router/batchrouter/asyncdestinationmanager/snowpipestreaming/internal/api/api_integration_test.go

```diff
@@ -139,25 +143,104 @@ func TestAPIIntegration(t *testing.T) {
+				enableCompression: false,
+				payloadSize:       839,
+			},
+		}
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				c := config.New()
+				c.Set("SnowpipeStreaming.enableCompression", tc.enableCompression)
+
+				statsStore, err := memstats.New()
+				require.NoError(t, err)
+
+				snowpipeAPI := api.New(c, statsStore, testConfig.clientURL, http.DefaultClient)
+
+				t.Log("Creating channel")
+				createChannelRes, err := snowpipeAPI.CreateChannel(ctx, &model.CreateChannelRequest{
+					RudderIdentifier: "1",
+					Partition:        "1",
+					AccountConfig: model.AccountConfig{
+						Account:              testConfig.credentials.Account,
+						User:                 testConfig.credentials.User,
+						Role:                 testConfig.credentials.Role,
+						PrivateKey:           strings.ReplaceAll(testConfig.credentials.PrivateKey, "\n", "\\\\\n"),
+						PrivateKeyPassphrase: testConfig.credentials.PrivateKeyPassphrase,
+					},
+					TableConfig: model.TableConfig{
+						Database: testConfig.credentials.Database,
+						Schema:   testConfig.namespace,
+						Table:    testConfig.tableName,
+					},
+				})
+				require.NoError(t, err)
+				require.NotEmpty(t, createChannelRes.ChannelID)
+				require.True(t, createChannelRes.Valid)
+				require.False(t, createChannelRes.Deleted)
+				require.EqualValues(t, whutils.ModelTableSchema{"ACTIVE": "boolean", "AGE": "int", "DOB": "datetime", "EMAIL": "string", "ID": "string", "NAME": "string"},
+					createChannelRes.SnowpipeSchema,
+				)
+
+				rows := []model.Row{
+					{"ID": "ID1", "NAME": "Alice Johnson", "EMAIL": "alice.johnson@example.com", "AGE": 28, "ACTIVE": true, "DOB": "1995-06-15T12:30:00Z"},
+					{"ID": "ID2", "NAME": "Bob Smith", "EMAIL": "bob.smith@example.com", "AGE": 35, "ACTIVE": true, "DOB": "1988-01-20T09:30:00Z"},
+					{"ID": "ID3", "NAME": "Charlie Brown", "EMAIL": "charlie.brown@example.com", "AGE": 22, "ACTIVE": false, "DOB": "2001-11-05T14:45:00Z"},
+					{"ID": "ID4", "NAME": "Diana Prince", "EMAIL": "diana.prince@example.com", "AGE": 30, "ACTIVE": true, "DOB": "1993-08-18T08:15:00Z"},
+					{"ID": "ID5", "NAME": "Eve Adams", "AGE": 45, "ACTIVE": true, "DOB": "1978-03-22T16:50:00Z"}, // -- No email
+					{"ID": "ID6", "NAME": "Frank Castle", "EMAIL": "frank.castle@example.com", "AGE": 38, "ACTIVE": false, "DOB": "1985-09-14T10:10:00Z"},
+					{"ID": "ID7", "NAME": "Grace Hopper", "EMAIL": "grace.hopper@example.com", "AGE": 85, "ACTIVE": true, "DOB": "1936-12-09T11:30:00Z"},
+				}
+
+				t.Log("Inserting records")
+				insertRes, err := snowpipeAPI.Insert(ctx, createChannelRes.ChannelID, &model.InsertRequest{
+					Rows:   rows,
+					Offset: "8",
+				})
+				require.NoError(t, err)
+				require.Equal(t, &model.InsertResponse{Success: true, Errors: nil}, insertRes)
+				require.EqualValues(t, tc.payloadSize, statsStore.Get("snowpipe_streaming_request_body_size", stats.Tags{
+					"api": "insert",
+				}).LastValue())
```

Comment:
we should also check if we are getting correct data after the compression.

Reply:
Good point. I was already doing it in "Create channel + Get channel + Insert data + Status" since the config is enabled by default. 
Let me modify the test so that is can do it for both compression enabled and disabled.